### PR TITLE
debian: Support downloading from a mirror in addition to having it locally

### PR DESCRIPTION
**Pull request by iainlane**
I tried not to break local mirror support, but I haven't actually tested that I didn't...

### CODE REVIEW

File: source/backends/debian/debpkg.d [hunk 1]
```diff
@@ -56,7 +58,16 @@ public:
     @property override const(string[string]) description () const { return desc; }
 
     override
-    @property string filename () const { return debFname; }
+    @property string filename () const {
+        if (debFname.isRemote) {
+            immutable path = buildPath (tmpDir, debFname.baseName);
+
+            downloadFile (debFname, path);
```
Comment (ximion) [inline source/backends/debian/debpkg.d]: Shouldn't that be `downloadIfNecessary`? Otherwise we might unnecessarily download multiple times...

Comment (iainlane) [inline source/backends/debian/debpkg.d]: `downloadFile` itself avoids re-downloading if the dest exists. Here I have the entire path, so it's easier to download directly; `downloadIfNecessary` is easier to use if you're constructing the path yourself from a base and suffix.

(That reminds me that `downloadIfNecessary`'s doc comment needs expanding...)

**Comment by ximion** [comment]
Looks good to me, I will merge it when you report that it worked well on Ubuntu.
So far I can't spot any obvious problems though, so I think this will be fine :-)

**Comment by iainlane** [comment]
It finished last night without crashing again (need to try the extra changes).

`ubuntu@asgen:~/appstream-workdir/export/data/yakkety$ zgrep -c "ID*" */Components-*.gz
main/Components-amd64.yml.gz:968
main/Components-arm64.yml.gz:961
main/Components-armhf.yml.gz:971
main/Components-i386.yml.gz:973
main/Components-powerpc.yml.gz:961
main/Components-ppc64el.yml.gz:961
main/Components-s390x.yml.gz:919
multiverse/Components-amd64.yml.gz:164
multiverse/Components-arm64.yml.gz:154
multiverse/Components-armhf.yml.gz:149
multiverse/Components-i386.yml.gz:168
multiverse/Components-powerpc.yml.gz:154
multiverse/Components-ppc64el.yml.gz:149
multiverse/Components-s390x.yml.gz:154
universe/Components-amd64.yml.gz:9673
universe/Components-arm64.yml.gz:9554
universe/Components-armhf.yml.gz:9547
universe/Components-i386.yml.gz:9663
universe/Components-powerpc.yml.gz:9551
universe/Components-ppc64el.yml.gz:9522
universe/Components-s390x.yml.gz:9361
`

**Comment by ximion** [comment]
Thanks a lot for working on this!

**Pull request merged**